af/fastpath/fastpath/db.py

```diff
@@ -156,3 +168,158 @@ def upsert_summary(
         # notification_json = ujson.dumps(notification)
         # q = f"SELECT pg_notify('fastpath', '{notification_json}');"
         # cur.execute(q)
+
+
+# # Clickhouse
+
+
+def query_click(query, query_params):
+    q = click_client.execute(query, query_params, with_column_types=True)
+    rows, coldata = q
+    colnames, coltypes = tuple(zip(*coldata))
+
+    for row in rows:
+        yield dict(zip(colnames, row))
+
+
+def _click_create_table_fastpath():
+    # TODO: table creation should be done before starting workers
+    sql = """
+    CREATE TABLE IF NOT EXISTS fastpath
+    (
+        `measurement_uid` String,
+        `report_id` String,
+        `input` String,
+        `probe_cc` String,
+        `probe_asn` Int32,
+        `test_name` String,
+        `test_start_time` DateTime,
+        `measurement_start_time` DateTime,
+        `filename` String,
+        `scores` String,
```

Comment:
Is the idea here to use the [clickhouse JSON functions](https://clickhouse.com/docs/en/sql-reference/functions/json-functions/) for interacting with the `scores` JSON string?

Did you consider the possibility of using a [Map](https://clickhouse.com/docs/en/sql-reference/data-types/map/) instead? If so why was the string data type chosen in place of Map?

Reply:
The JSON structure is nested.